Release notes v0.6.5 (#853)
napari/docs: docs/release/release_0_6_5.md

For more information, examples, and documentation, please visit our website,
https://napari.org.

napari follows [EffVer (Intended Effort Versioning)](https://effver.org/); this is a **Meso** release containing awesome new features, but some effort may be needed when updating previous projects to use this version.

> 

Reviewer: the script I wrote introduces extra line breaks before and after here, and I couldn't figure out why and forgot to ask. We can remove them or keep them, doens't matter

## Highlights

### Define a startup script for custom launch behaviour

> Do you have a code snippet that you always find yourself running after you launch napari? No more! You can now put this code in a script and set its path in the new `startup script` setting ([#8188](https://github.com/napari/napari/pull/8188)), and it will be executed every time napari opens. It's just a python script, so sky's the limit :) We found it particularly useful for adding custom colormaps, setting up the scale bar *just right*, or automatically launching our favourite plugin on startup.

Reviewer: ```suggestion
Do you have a code snippet that you always find yourself running after you launch napari? No more! You can now put this code in a script and set its path in the new `startup script` setting ([#8188](https://github.com/napari/napari/pull/8188)), and it will be executed every time napari opens. It's just a python script, so sky's the limit :) We find it particularly useful for adding custom colormaps, setting up the scale bar *just right*, or automatically launching our favourite plugin on startup.
```

Author: Isn't it wrong like this? "We" found it useful to launch "your" favourite plugin? xD

Reviewer: oops you are correct. I edited it now to switch the verb tense from found to find. 
(before it was saying "our favourite" and I wrongly did "your favourite")

TODO: image here

### Automatically tiled overlays and ColorBar overlay
Canvas overlays such as `scale_bar`, `text_overlay`, and `color_bar` overlay are now automatically tiling ([#7836](https://github.com/napari/napari/pull/7836)), preventing annoying overlap and making them easier to use without having to manage positioning. Wait, `color_bar` overlay you said? You heard it right! This is a new overlay ([#7832](https://github.com/napari/napari/pull/7832))that shows a color bar legend, and it works with any layer which uses a colormap. All of this works seamlessly with multiple overlays and even grid mode:

```py
import napari

viewer = napari.Viewer()

# enable grid with stride 2 to get layers split two-by-two
viewer.grid.enabled = True
viewer.grid.stride = 2

# set the scale bar to gridded mode so it appears in each grid box
viewer.scale_bar.visible = True
viewer.scale_bar.gridded = True

layers = viewer.open_sample('napari', 'lily')

# enable color bars
for layer in layers:

>     layer.color_bar.visible = True

Reviewer: FWIW, I do think this is totally fine for public API.
Should it be `colorbar` or `color_bar`?

Author: I think this should be discussed in github.com/napari/napari/pull/7832!

```

> TODO: image here

Author: ```suggestion
![Image depicting the napari viewer in grid mode with scale bars and color bars enabled](https://private-user-images.githubusercontent.com/23482191/494350517-622b2d36-11a7-4c55-9550-c82ddebc2fda.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg4ODU5NzksIm5iZiI6MTc1ODg4NTY3OSwicGF0aCI6Ii8yMzQ4MjE5MS80OTQzNTA1MTctNjIyYjJkMzYtMTFhNy00YzU1LTk1NTAtYzgyZGRlYmMyZmRhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTI2VDExMjExOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjODA3ZmExZTY4NzA3NzQ3YzQ3MGY2NzM1MDg4MmFhYWM2ZTA0MjIyMDBlMTE1Y2NjMGYxMmFiYzZlMGVmNTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-xXGxQm1VvTSJMQj0t2IReHhjbT-lSt8oxobpru7Px0)
```

> ![Image depicting the napari viewer in grid mode with scale bars and color bars enabled](https://private-user-images.githubusercontent.com/23482191/494350517-622b2d36-11a7-4c55-9550-c82ddebc2fda.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg4ODU5NzksIm5iZiI6MTc1ODg4NTY3OSwicGF0aCI6Ii8yMzQ4MjE5MS80OTQzNTA1MTctNjIyYjJkMzYtMTFhNy00YzU1LTk1NTAtYzgyZGRlYmMyZmRhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTI2VDExMjExOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjODA3ZmExZTY4NzA3NzQ3YzQ3MGY2NzM1MDg4MmFhYWM2ZTA0MjIyMDBlMTE1Y2NjMGYxMmFiYzZlMGVmNTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-xXGxQm1VvTSJMQj0t2IReHhjbT-lSt8oxobpru7Px0)

Reviewer: so I think this link won't work because its 'private-user-images' at the beginning. Theres a different way to get the public link. I couldn't remember exactly how to do it so it isn't in the guide. TBH I just click around a bunch

Author: I can't get any other link than this kind. Did you anything different from just uploading and copy pasting the image in the issue? Is it a format thing maybe?

Reviewer: ok I'll work on figure it out :)

Reviewer: Here is an example of what it needs to look like
https://github.com/napari/docs/blob/7aea91ad0586833ce9b04f76a8c002943a5d8a60/docs/release/release_0_6_3.md?plain=1#L17C1-L17C126
`![GIF Displaying Alt-Drag Zoom Box Behavior](https://github.com/user-attachments/assets/f32ea020-28e2-4059-90b9-491bdd4a962b)`
If you right click the image in a comment, you get the `private` path. 
Instead, if you "edit" the comment you'll see the public path

Reviewer: This should be the image for colorbars, I think `https://github.com/user-attachments/assets/622b2d36-11a7-4c55-9550-c82ddebc2fda`

Reviewer: Note how if you click your private link (at least from a private browser, I dunno if logged in) it 404s, but with the public you can open from anywhere

Author: I'm sure I tried the "edit" thing :thinking: I must have done something else xD

> ### Task manager will now try to save your unfinished work

Reviewer: I dont mind the title here, but save can have two meanings "rescue" and "place on drive". at first I read it as "store on the drive" which is definitely NOT what we want people to think. 
Maybe the below, but it kind of seems too technical?
```suggestion
### Task manager will now try to protect events running in separate threads
```

Author: True, but `events` is also very loaded in napari context. What about "try to prevent losing unfinished work"?

Reviewer: I like that!

> We added a new task manager ([#8211](https://github.com/napari/napari/pull/8211)) which automatically register any running `thread_worker`, showing a confirmation dialog if you attempt to close napari while a task is running.

Reviewer: ```suggestion
We added a new task manager ([#8211](https://github.com/napari/napari/pull/8211)) which automatically registers any running `thread_worker`, showing a confirmation dialog if you attempt to close napari while a task is running.
```

### New `remove()` and `pop()` methods for Points and Shapes
Points and Shapes can now be easily removed, not just added :P ([#8031](https://github.com/napari/napari/pull/8031) and [#8072](https://github.com/napari/napari/pull/8072)).

### A new and updated guide on napari Preferences
Our documentation on the napari Preferences has received a major overhaul! [Check it out here](preferences).

> TODO: does this link work?

Reviewer: this _would_ work on the docs, but it won't work on the Github Release Notes version, so you'll want to hard code the link to dev version of page (for pre-release, at least)
